pkg/unikontainers/config_test.go

```diff
@@ -52,24 +52,24 @@ func TestGetConfigFromSpec(t *testing.T) {
 			MountRootfs:     "true",
 		}
 
-		config, err := getConfigFromSpec(spec)
-		assert.NoError(t, err, "Expected no error")
+		config := getConfigFromSpec(spec)
 		assert.Equal(t, expectedConfig, config, "Expected config to match")
+		err := config.validate()
+		assert.NoError(t, err, "Expected a full config to be valid")
 	})
 
-	t.Run("get config from spec empty annotations", func(t *testing.T) {
+	t.Run("get config from spec with empty annotations", func(t *testing.T) {
 		t.Parallel()
 		spec := &specs.Spec{
 			Annotations: map[string]string{},
 		}
-
-		config, err := getConfigFromSpec(spec)
-		assert.Error(t, err, "Expected an error")
-		assert.Nil(t, config, "Expected config to be nil")
-		assert.Equal(t, ErrEmptyAnnotations, err, "Expected ErrEmptyAnnotations")
+		config := getConfigFromSpec(spec)
+		assert.NotNil(t, config, "Expected config to be non-nil even with empty annotations")
+		err := config.validate()
+		assert.Error(t, err, "Expected validation to fail for an empty config")
```

Comment:
Could you add one more check using https://pkg.go.dev/github.com/stretchr/testify/assert#ErrorContains and compare the error text returned?